include/Operator/SecondQuantized/USQHamiltonian.hpp

```diff
@@ -0,0 +1,250 @@
+// This file is part of GQCG-gqcp.
+// 
+// Copyright (C) 2017-2019  the GQCG developers
+// 
+// GQCG-gqcp is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#pragma once
+
+
+#include "Operator/SecondQuantized/SQHamiltonian.hpp"
+
+
+namespace GQCP {
+
+
+/**
+ *  A class for representing the unrestricted second-quantized electronic Hamiltonian
+ *  It consists of two restricted Hamiltonians for both the beta and alpha component and a mixed two-electron term
+ * 
+ *
+ *  @tparam Scalar      the scalar type of the second-quantized parameters (i.e. the integrals)
+ */
+template <typename Scalar>
+class USQHamiltonian {
+private:
+    SQHamiltonian<Scalar> sq_hamiltonian_alpha;  // alpha hamiltonian
+    SQHamiltonian<Scalar> sq_hamiltonian_beta;  // beta hamiltonian
+
+    std::vector<ScalarSQTwoElectronOperator<Scalar>> two_op_mixed;  // alpha & beta mixed operators represented as g_aabb
+    ScalarSQTwoElectronOperator<Scalar> total_two_op_mixed;  // the total alpha & beta mixed operator represented as g_aabb
+
+public:
+
+    /*
+     *  CONSTRUCTORS
+     */
+    USQHamiltonian() = default;
+
+    /**
+     *  @param sq_hamiltonian_alpha      alpha hamiltonian
+     *  @param sq_hamiltonian_beta       beta hamiltonian
+     *  @param two_op_mixed              alpha & beta mixed operators
+     */
+    USQHamiltonian(const SQHamiltonian<Scalar>& sq_hamiltonian_alpha, const SQHamiltonian<Scalar>& sq_hamiltonian_beta, const std::vector<ScalarSQTwoElectronOperator<Scalar>>& two_op_mixed) :
+        sq_hamiltonian_alpha (sq_hamiltonian_alpha),
+        sq_hamiltonian_beta (sq_hamiltonian_beta),
+        two_op_mixed (two_op_mixed)
+    {
+
+        const auto dim = sq_hamiltonian_alpha.dimension();
```

Comment:
Add 'header comment'
```suggestion

        // Check if the dimensions are compatible
        const auto dim = sq_hamiltonian_alpha.dimension();
```